src/Packetery/Module/Checkout/Checkout.php

```diff
@@ -0,0 +1,385 @@
+		$this->wpAdapter->addAction(
+			'woocommerce_store_api_checkout_order_processed',
+			[
+				$this->orderUpdater,
+				'actionUpdateOrder',
+			]
+		);
+
+		// Must not be registered at backend.
+		$this->wpAdapter->addFilter( 'woocommerce_available_payment_gateways', [ $this, 'filterPaymentGateways' ] );
+
+		$this->wpAdapter->addAction(
+			'woocommerce_review_order_before_shipping',
+			[
+				$this->sessionService,
+				'actionUpdateShippingRates',
+			]
+		);
+		$this->wpAdapter->addFilter(
+			'woocommerce_cart_shipping_packages',
+			[
+				$this->sessionService,
+				'filterUpdateShippingPackages',
+			]
+		);
+		$this->wpAdapter->addAction( 'woocommerce_cart_calculate_fees', [ $this, 'actionCalculateFees' ] );
+		$this->wpAdapter->addAction(
+			'init',
+			function () {
+				/**
+				 * Tells if widget button table row should be used.
+				 *
+				 * @since 1.3.0
+				 */
+				if ( $this->optionsProvider->getCheckoutWidgetButtonLocation() === 'after_transport_methods' ) {
+					$this->wpAdapter->addAction(
+						'woocommerce_review_order_after_shipping',
+						[
+							$this->renderer,
+							'actionRenderWidgetButtonTableRow',
+						]
+					);
+				} else {
+					$this->wpAdapter->addAction(
+						'woocommerce_after_shipping_rate',
+						[
+							$this->renderer,
+							'actionRenderWidgetButtonAfterShippingRate',
+						]
+					);
+				}
+			}
+		);
+
+		$this->wpAdapter->addAction(
+			'woocommerce_review_order_after_shipping',
+			[
+				$this->renderer,
+				'actionRenderEstimatedDeliveryDateSection',
+			]
+		);
+	}
+
+	/**
+	 * Calculates fees.
+	 *
+	 * @return void
+	 * @throws ProductNotFoundException Product not found.
+	 */
+	public function actionCalculateFees(): void {
+		$chosenShippingMethod = $this->checkoutService->calculateShippingAndGetId();
+		if ( false === $this->checkoutService->isPacketeryShippingMethod( $chosenShippingMethod ) ) {
+			return;
+		}
+
+		$carrierOptions = $this->carrierOptionsFactory->createByOptionId( $chosenShippingMethod );
+		$chosenCarrier  = $this->carrierEntityRepository->getAnyById( $this->checkoutService->getCarrierIdFromShippingMethod( $chosenShippingMethod ) );
+		$maxTaxClass    = $this->cartService->getTaxClassWithMaxRate();
+
+		if (
+			$carrierOptions->hasCouponFreeShippingForFeesAllowed() &&
+			$this->rateCalculator->isFreeShippingCouponApplied( $this->wcAdapter->cart() )
+		) {
+			return;
+		}
+
+		if (
+			null !== $chosenCarrier &&
+			$chosenCarrier->supportsAgeVerification() &&
+			null !== $carrierOptions->getAgeVerificationFee() &&
+			$this->cartService->isAgeVerification18PlusRequired()
+		) {
+			$feeAmount = $this->currencySwitcherService->getConvertedPrice( $carrierOptions->getAgeVerificationFee() );
+			if ( false !== $maxTaxClass && $feeAmount > 0 && $this->optionsProvider->arePricesTaxInclusive() ) {
+				$feeAmount = $this->calcTaxExclusiveFeeAmount( $feeAmount, $maxTaxClass );
+			}
+
+			$this->wcAdapter->cartFeesApiAddFee(
+				[
```

Comment:
nemělo by to být DTO? to samé se používá i na konci této funkce

Reply:
nebo při nejmenším nějak sjednotit

Reply:
Jako že bych to pole předal konstruktoru, abych na další řádce volal `get_object_vars` abych to mohl předat WC?

Reply:
Nebo spíš něco takového 

`private function createFeeProperties( string $id, string $name, float $amount, bool $isTaxable, ?string $taxClass ): array {
    return [
        'id'        => $id,
        'name'      => $name,
        'amount'    => $amount,
        'taxable'   => $isTaxable,
        'tax_class' => $taxClass,
    ];
}`